color-changer.js

```diff
@@ -0,0 +1,42 @@
+function getRandomUnusedHue(usedHues) {
+    let randomHue;
+    while(randomHue === undefined || usedHues[randomHue]) {
+        randomHue = Math.floor(Math.random()*360);
+    }
+    return randomHue;
+}
+
+function changeBgHue(newHue) {
+    let minLightness = 30;
+    let maxLightness = 70;
+    const randomLightness = Math.floor(Math.random() * (maxLightness - minLightness) + minLightness)
+    document.body.style.backgroundColor = `hsl(${newHue}, 50%, ${randomLightness}%)`;
+}
+
+function changeBgColorRandomly() {
+    const usedHues = {};
+      
+    let randomBgColorChange = function() {
+        let newHue = getRandomUnusedHue(usedHues);
+        changeBgHue(newHue)
+        usedHues[newHue] = true;
+    }
+
+    let count=1;
+    randomBgColorChange();
+    count++;
```

Comment:
It's up for debate how closely I followed the Single Responsibility Principle in this function overall, but I wanted randomBgColorChange to be about changing to a random hue. Counter logic doesn't belong in hue changing. Seemed more appropriate inside the setInterval callback where I'm actually using the counter to determine if I should continue.